janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -498,6 +506,117 @@ private static Query numericQuery(String key, Cmp relation, Number value) {
         }
     }
 
+    // adapted from SolrIndex
+    private List<String> customTokenize(Analyzer analyzer, String fieldName, String value) {
+        CachingTokenFilter stream = null;
+        try {
+            final List<String> terms = new ArrayList<>();
+            stream = new CachingTokenFilter(analyzer.tokenStream(fieldName, value));
+            final TermToBytesRefAttribute termAtt = stream.getAttribute(TermToBytesRefAttribute.class);
+            stream.reset();
+            while (stream.incrementToken()) {
+                terms.add(termAtt.getBytesRef().utf8ToString());
+            }
+            return terms;
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        } finally {
+            org.apache.commons.io.IOUtils.closeQuietly(stream);
+        }
+    }
+
+    private void tokenize(SearchParams params, KeyInformation.StoreRetriever information, String value, String key, JanusGraphPredicate janusgraphPredicate) {
+        final Parameter[] parameters = information.get(key).getParameters();
+        Mapping mapping = ParameterType.MAPPING.findParameter(parameters, null);
+
+        final String analyzerString;
+        if (mapping != null) switch (mapping) {
+            case STRING:
+                analyzerString = ParameterType.STRING_ANALYZER.findParameter(parameters, null);
+                break;
+            case DEFAULT:
+            case TEXT:
+                analyzerString = ParameterType.TEXT_ANALYZER.findParameter(parameters, null);
+                break;
+            default:
+                throw new RuntimeException("unsupported case " + mapping);
+        } else analyzerString = null;
+
+        List<String> terms;
+        if (analyzerString != null) {
+            try {
+                final Class classDefinition = Class.forName(analyzerString);
+                final Analyzer analyzer = (Analyzer) classDefinition.newInstance();
+                terms = customTokenize(analyzer, key, value);
+            } catch (Exception e) {
+                throw new RuntimeException("Analyzer cannot be instanciated for class " + analyzerString, e);
+            }
+
+        } else {
+            if (mapping == null) mapping = Mapping.DEFAULT;
```

Comment:
Define conditional on multiple lines with curly braces (for consistency with code style recommendations in other code reviews in the project).